dot/network/inbound.go

```diff
@@ -0,0 +1,73 @@
+package network
+
+import (
+	libp2pnetwork "github.com/libp2p/go-libp2p-core/network"
+)
+
+func (s *Service) readStream(stream libp2pnetwork.Stream, decoder messageDecoder, handler messageHandler) {
+	// we NEED to reset the stream if we ever return from this function, as if we return,
+	// the stream will never again be read by us, so we need to tell the remote side we're
+	// done with this stream, and they should also forget about it.
+	defer s.resetInboundStream(stream)
+	s.streamManager.logNewStream(stream)
+
+	peer := stream.Conn().RemotePeer()
+	msgBytes := s.bufPool.get()
```

Comment:
why do we need a different pool?

Reply:
Because each buffer cap might be different here than the buffer cap in the previous (single) use case of this pool. In example [this](https://github.com/ChainSafe/gossamer/blob/qdm12-mem-hasher-fix/lib/trie/hash.go#L31-L45) so that we don't use large 1MB buffers for 32 bytes mem operation.


If it's around the same then no point using a different one.

Reply:
yeah, it kind of depends on the message type I guess. `BlockAnnounce`s should generally be pretty small, but `ConsensusMessage`s and `TransactionMessage`s could be quite large. I could open an issue to explore this more

Reply:
#2045

Reply:
Nice :+1: That might be a micro optimization, let's not waste time on it yet. Thanks for creating the issue!